src/uct/cuda/cuda_ipc/cuda_ipc_md.c

```diff
@@ -96,16 +192,36 @@ static ucs_status_t uct_cuda_ipc_query_md_resources(uct_md_resource_desc_t **res
     return uct_single_md_resource(&uct_cuda_ipc_md_component, resources_p, num_resources_p);
 }
 
+static int uct_is_cuda_ipc_mem_type_owned(uct_md_h md, void *addr, size_t length)
+{
+    int      memory_type;
+    CUresult cu_ret;
+
+    if (addr == NULL) {
+        return 0;
+    }
+
+    cu_ret = cuPointerGetAttribute(&memory_type,
+                                   CU_POINTER_ATTRIBUTE_MEMORY_TYPE,
+                                   (CUdeviceptr)addr);
+    if (cu_ret == CUDA_SUCCESS) {
+        if (memory_type == CU_MEMORYTYPE_DEVICE) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
```

Comment:
there is common code already, https://github.com/openucx/ucx/blob/master/src/uct/cuda/base/cuda_md.c#L11